File: InMotionGIT.Common/Services/Contracts/DataCommand.vb
﻿Imports System.Runtime.Serialization
Imports System.Xml.Serialization
Imports InMotionGIT.Common.DataType

Namespace Services.Contracts

    <DataContract()>
    Public Class DataCommand

        <DataMember()>
        Public ConnectionStringName As String

        <DataMember()>
        Public Property TableName As String

        <DataMember()>
        Public Property Statement As String

        <DataMember()>
        Public Property Operation As String

        <DataMember()>
        <XmlArray(ElementName:="Fields")>
        <XmlArrayItem(GetType(Dictionary(Of String, String)))>
        Public Property Fields As Dictionary(Of String, String) = New Dictionary(Of String, String)

        <DataMember()>
        Public Property Parameters As List(Of DataParameter)

        <DataMember()>
        Public Property Owner As String

        <DataMember()>
        Public Property ObjectType As String

        <DataMember()>
        Public Property CompanyId As Integer

        <DataMember()>
        Public Property ConnectionStringsRaw As Services.Contracts.ConnectionString

        <DataMember()>
        Public Property MaxNumberOfRecord As Integer

        <DataMember()>
        Public Property IgnoreMaxNumberOfRecords As Boolean

        <DataMember()>
        Public Property QueryCount As String

        <DataMember()>
        Public Property QueryCountResult As Integer

        <DataMember()>
        Public Property LookUp As LookUpValue

    End Class

End Namespace

File: Common/InMotionGIT.Common.Domain/DataAcceess/DataCommand.cs
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Xml.Serialization; 

namespace InMotionGIT.Common.Domain.DataAccess;


[DataContract()]
public class DataCommand
{

    [DataMember()]
    public string ConnectionStringName;

    [DataMember()]
    public string TableName { get; set; }

    [DataMember()]
    public string Statement { get; set; }

    [DataMember()]
    public string Operation { get; set; }

    [DataMember()]
    [XmlArray(ElementName = "Fields")]
    [XmlArrayItem(typeof(Dictionary<string, string>))]
    public Dictionary<string, string> Fields { get; set; } = new Dictionary<string, string>();

    [DataMember()]
    public List<DataParameter> Parameters { get; set; }

    [DataMember()]
    public string Owner { get; set; }

    [DataMember()]
    public string ObjectType { get; set; }

    [DataMember()]
    public int CompanyId { get; set; }

    [DataMember()]
    public ConnectionStrings ConnectionStringsRaw { get; set; }

    [DataMember()]
    public int MaxNumberOfRecord { get; set; }

    [DataMember()]
    public bool IgnoreMaxNumberOfRecords { get; set; }

    [DataMember()]
    public string QueryCount { get; set; }

    [DataMember()]
    public int QueryCountResult { get; set; }

    [DataMember()]
    public DataType. LookUpValue LookUp { get; set; }

}